FrEIA/modules/all_in_one_block.py

```diff
@@ -155,11 +159,14 @@ class or callable ``f``, called as ``f(channels_in, channels_out)`` and
             self.w_perm = None
             self.w_perm_inv = None
             self.w_0 = nn.Parameter(torch.FloatTensor(w), requires_grad=False)
-        else:
+        elif permute_soft:
             self.w_perm = nn.Parameter(torch.FloatTensor(w).view(channels, channels, *([1] * self.input_rank)),
                                        requires_grad=False)
             self.w_perm_inv = nn.Parameter(torch.FloatTensor(w.T).view(channels, channels, *([1] * self.input_rank)),
                                            requires_grad=False)
```

Comment:
While we're here: Isn't `torch.from_numpy` preferred over tensor(array)?

Reply:
Also a bit strange that the parameter is being saved as a view. A `reshape` or `view().contiguous()` seems more appropriate, unless the `view` has a specific purpose?

Reply:
I tested the suggested changes, and they don't seem to violate any of my local or PR pipepline testing. `torch.from_numpy` loses the explicit typecast to float, and instead uses double (as the scipy default). I personally think this is unnecessary and if you agree I propose revert this change to `torch.FloatTensor`. as for `view().contiguous()` I see no reason this should not be included.